tests/migration.go

```diff
@@ -72,6 +77,80 @@ func ConfirmVMIPostMigration(virtClient kubecli.KubevirtClient, vmi *v1.VirtualM
 
 	By("Verifying the VMI's is in the running state")
 	Expect(vmi.Status.Phase).To(Equal(v1.Running), "the VMI must be in `Running` state after the migration")
+
+	return vmi
+}
+
+func setOrClearDedicatedMigrationNetwork(nad string, set bool) *v1.KubeVirt {
+	virtClient, err := kubecli.GetKubevirtClient()
+	Expect(err).ToNot(HaveOccurred())
+
+	kv := util.GetCurrentKv(virtClient)
+
+	// Saving the list of virt-handler pods prior to changing migration settings, see comment below.
+	listOptions := metav1.ListOptions{LabelSelector: v1.AppLabel + "=virt-handler"}
+	virtHandlerPods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(), listOptions)
+	Expect(err).ToNot(HaveOccurred(), "Failed to list the virt-handler pods")
+
+	if set {
+		if kv.Spec.Configuration.MigrationConfiguration == nil {
+			kv.Spec.Configuration.MigrationConfiguration = &v1.MigrationConfiguration{}
+		}
+		kv.Spec.Configuration.MigrationConfiguration.Network = &nad
+	} else {
+		if kv.Spec.Configuration.MigrationConfiguration != nil {
+			kv.Spec.Configuration.MigrationConfiguration.Network = nil
+		}
+	}
+
+	res := UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+	// By design, changing migration settings trigger a re-creation of the virt-handler pods, amongst other things.
+	//   However, even if SetDedicatedMigrationNetwork() calls UpdateKubeVirtConfigValueAndWait(), VMIs still seem to get scheduled on outdated virt-handler pods.
+	//   Not sure if it's a bug or a feature though. Waiting for all "old" virt-handlers to disappear ensures test VMIs will be created on updated virt-handler pods.
+	Eventually(func() bool {
+		for _, pod := range virtHandlerPods.Items {
+			_, err = virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			if err == nil {
+				return false
+			}
+		}
+		return true
+	}, 180*time.Second, 10*time.Second).Should(BeTrue(), "Some virt-handler pods survived the migration settings change")
+
+	return res
```

Comment:
I think we need to check the new handlers are ready otherwise we will see this failure https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6036/pull-kubevirt-e2e-k8s-1.20-sig-network/1468980795975667712